vividus/src/main/java/org/vividus/StatisticsStoryReporter.java

```diff
@@ -256,67 +223,57 @@ public void afterStoriesSteps(Stage stage)
         }
     }
 
-    private void calculateStatus(Node node)
+    private Node startNode(NodeType type)
     {
-        if (hasNoExecutedChildrens(node))
-        {
-            node.withStatus(Status.SKIPPED);
-            aggregate(node);
-            return;
-        }
-        node.getChildren().stream()
-                          .map(n ->
-                          {
-                              if (n.getType() == NodeType.STEP)
-                              {
-                                  aggregate(n);
-                              }
-                              return n;
-                          })
-                          .reduce((l, r) -> l.getStatus().getPriority() < r.getStatus().getPriority() ? l : r)
-                          .map(Node::getStatus)
-                          .map(node::withStatus)
-                          .ifPresentOrElse(this::aggregate, () -> {
-                              node.withStatus(Status.PASSED);
-                              aggregate(node);
-                          });
+        Node node = new Node(type);
+        context().addToTail(node);
+        return node;
     }
 
-    private boolean hasNoExecutedChildrens(Node node)
+    private void updateStepStatus(Status status)
     {
-        return node.getChildren().isEmpty() && node.isHasChildrens();
+        updateNodeStatus(context().getTail(), status);
     }
 
-    private void aggregate(Node node)
+    private void endStep(Status status)
     {
-        Statistic stat = AGGREGATOR.get(node.getType());
-        mapper.get(node.getStatus()).accept(stat);
+        updateStepStatus(status);
+        endNode();
     }
 
-    private void updateStepStatus(Status status)
+    private void endNode()
     {
-        NodeContext context = context();
-        Node currentNode = isRoot() ? context.getRoot() : context.getTail();
-        Iterator<Node> iterator = currentNode.getChildren().descendingIterator();
-        while (iterator.hasNext())
+        Node currentNode = context().getTail();
+        if (currentNode.getChildren().isEmpty() && currentNode.isHasChildren())
```

Comment:
when it might happen?

Reply:
empty scenario, I thought you are an author of this code :)